Xamarin.PropertyEditing/Numeric.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Reflection.Emit;
+using System.Threading.Tasks;
+
+namespace Xamarin.PropertyEditing
+{
+	internal static class Numeric<T>
+	{
+		public static Func<T, T> Increment
+		{
+			get
+			{ 
+				SetupMath();
+				return IncrementValue;
+			}
+		}
+
+		public static Func<T, T> Decrement
+		{
+			get
+			{
+				SetupMath();
+				return DecrementValue;
+			}
+		}
+
+		private static Func<T, T> IncrementValue, DecrementValue;
+
+		private static void SetupMath ()
+		{
+			if (IncrementValue != null)
+				return;
+
+			Task<Func<T, T>> createIncrement = Task.Run (() => {
```

Comment:
Hehe, you knew I'd comment on that. In addition to the pattern matching solution that's been suggested elsewhere, I don't see why you shouldn't also be able to setup the increment and decrement delegates early on to point to Lambdas written in plain C# that are specialized for each type, but can be cast to your generic signature.
Then again, as far as IL gen goes, this is not the worst ;)

Reply:
Frankly it was because I thought I could get away with fewer branches than this originally and that'd make it more concise than a full method for every type. Mainly just stayed on the train of knowing the approach would work one way or another.